client-lib/src/main/java/com/ibm/ws/repository/resources/internal/RepositoryResourceImpl.java

```diff
@@ -161,10 +162,16 @@ private RepositoryWriteableClient getWritableClient() throws RepositoryOperation
         if (_client instanceof RepositoryWriteableClient) {
             return (RepositoryWriteableClient) _client;
         } else {
-            throw new RepositoryOperationNotSupportedException("The backend does not support write operations", _repoConnection);
+            throw new RepositoryOperationNotSupportedException("The backend does not support write operations or no connection has been specified", getRepositoryConnection());
```

Comment:
It doesn't look like the new message matches up with the code. I can't see a check for whether the connection has been specified or not.